util/package/packagehelpers.rb

```diff
@@ -97,6 +97,32 @@ def self.yum_clean(clean_mode)
           return result
         end
 
+        def self.yum_isavailable(package, version)
+          raise "Cannot find yum at /usr/bin/yum" unless File.exists?("/usr/bin/yum")
+          result = {:exitcode => nil,
+                    :output => ""}
+
+          if version.nil?
+            cmd = Shell.new(
+              "/usr/bin/yum list -q #{package}",
+              :stdout => result[:output]
+            )
+          else
+            # If we specify the version, but the package would be downgraded, the information
+            # won't show without passing in --showduplicates
+            cmd = Shell.new(
+              "/usr/bin/yum list -q --showduplicates #{package}-#{version}",
+              :stdout => result[:output]
+            )
+          end
+
+          cmd.runcommand
+          result[:exitcode] = cmd.status.exitstatus
+
+          raise "Error: No matching Packages to list" unless result[:exitcode] == 0
```

Comment:
hmm, yeah if its call `yum_isavailable` then it should really have a boolean and not string or error, we can include the output as now but a boolean output variable on the action that answers the basic yes/no would be good?

Reply:
@ripienaar I'll probably rename this to `yum_list` then, it makes more sense than `yum_isavailable`.

Reply:
Would be good if this was something like `check_available` and it would call a correct package specific version underneath? Similar to the check update one?